public/app/plugins/datasource/graphite/FunctionEditor.tsx

```diff
@@ -68,6 +68,11 @@ class FunctionEditor extends React.PureComponent<FunctionEditorProps, FunctionEd
                   onClick={popperProps.show ? hidePopper : showPopper}
                   style={{ cursor: 'pointer' }}
                 >
+                  {this.props.func.def.unknown && (
```

Comment:
NIT: I like the idea of showing a warning to the user in this case, but wouldn't this clutter the interface a bit too much?

what about something like (pseudocode): 

```tsx
<span 
    ref={this.triggerRef}
    onClick={popperProps.show ? hidePopper : showPopper}
    style={{ cursor: 'pointer' }}
    className={cx(this.props.func.def.unknown && 'a_slighlty_different_color_that_indicates_a_warning'}
>
  {this.props.func.def.unknown ? (
    <Tooltip content="This function is not on the list of supported functions." placement="bottom">
      {this.props.func.def.name}
    </Tooltip>) : this.props.func.def.name }
</span>
```

Reply:
Because I'm color-blind and I need something else 😭 I was thinking about wavy underline but wasn't sure. I'll run it by UX.

Reply:
Good point!

btw, i'm not even sure it's needed since if the function is not defined you get an error anyway. but better loop UX in 👍

Reply:
I've added some changes suggested by UX:
- red border around the label
- padding between the icon and the label
- more descriptive tooltip content

> (...) btw, i'm not even sure it's needed since if the function is not defined you get an error anyway (...)

This is still useful in cases where there are multiple queries/functions. The error message contains only one unsupported function even if there are many:

<img width="551" alt="Screenshot 2021-04-06 at 16 11 22" src="https://user-images.githubusercontent.com/745532/113724556-c1a4ff00-96f2-11eb-832f-4dc8ffc3017f.png">